### PR TITLE
QC report: add lanes and sex ploidies plots, use metadata bucket

### PR DESCRIPTION
**Pull request by vladsavelyev**
Example: https://test-web.populationgenomics.org.au/tob-wgs/qc/qc-v2.html

### CODE REVIEW

File: reports/qc/script-for-analysis-runner.sh [hunk 1]
```diff
@@ -24,46 +28,41 @@ case $key in
 esac
 done
 
-micromamba install -y --prefix $MAMBA_ROOT_PREFIX -y -c conda-forge \
-	r-reactable r-ggrepel r-sessioninfo r-gargle r-here r-assertthat r-dt r-googlecloudstorager r-ggforce
+if [[ $ANALYSIS_RUNNER = "YES" ]] ; then
```
Comment (pdiakumis) [inline reports/qc/script-for-analysis-runner.sh:31]: Does this need a `==` or is bash just weird?

File: reports/qc/script-for-analysis-runner.sh [hunk 2]
```diff
@@ -24,46 +28,41 @@ case $key in
...
-
-	test -f ${dir}/gender.tsv || gsutil cp "gs://cpg-tob-wgs-test/gender.tsv" ${dir}/gender.tsv
-	test -f ${dir}/age.csv    || gsutil cp "gs://cpg-tob-wgs-test/age.csv" ${dir}/age.csv
-	test -f ${dir}/qc.csv     || gsutil cp "gs://cpg-tob-wgs-${main_suf}/gvcf/${batch}/*.csv" ${dir}/qc.csv
-	test -f ${dir}/meta.tsv   || gsutil cp "gs://cpg-tob-wgs-${main_suf}/joint-calling/${joint_calling_run_version}/meta.tsv" ${dir}/meta.tsv
+	test -f ${dir}/gender.tsv || gsutil cp "gs://cpg-tob-wgs-main-metadata/gender.tsv" ${dir}/gender.tsv
+	test -f ${dir}/age.csv    || gsutil cp "gs://cpg-tob-wgs-main-metadata/age.csv" ${dir}/age.csv
+	test -f ${dir}/qc.csv     || gsutil cp "gs://cpg-tob-wgs-main-metadata/${batch}/*.csv" ${dir}/qc.csv
+	test -f ${dir}/meta.tsv   || gsutil cp "gs://cpg-tob-wgs-main-metadata/joint-calling/${joint_calling_run_version}/meta.tsv" ${dir}/meta.tsv
+	cat qc.Rmd | sed 's/r fig.width=[1-9]*, fig.height=[1-9]*/r fig.width=plot_width, fig.height=plot_height/g' > qc-for-html.Rmd
```
Comment (pdiakumis) [inline reports/qc/script-for-analysis-runner.sh:46]: I don't get why you're sedding here and in the test script but I guess it's for a good reason

File: reports/qc/qc.Rmd [hunk 3]
```diff
@@ -510,6 +567,31 @@ all_data %>% filter(batch == "3", duplicates > 0.15) %>% dplyr::pull(sample)
...
+
+X ploidy vs Y ploidy. Samples with matched reported and inferred sex is labelled.
+
+```{r}
+sex_data %>% ggplot(aes(x = chrX_ploidy, y = chrY_ploidy, color = gender)) + 
+  geom_point(size = 0.6) +
+  geom_text_repel(aes(label = label), color = "black", na.rm = TRUE, size = 2.3)
+```
+
+Y ploidy vs age. Samples with matched reported and inferred sex is labelled. chY degrades with age, so low chrY ploidies for older males are expected (though not high chrX ploidies).
```
Comment (pdiakumis) [inline reports/qc/qc.Rmd]: Do you mean _mismatched_ samples are labelled?

**Review by pdiakumis — CHANGES_REQUESTED** [review]

File: reports/qc/script-for-analysis-runner.sh (hunk 1 above)
Comment (vladsavelyev) [reply reports/qc/script-for-analysis-runner.sh:31]: Bash is weird, yes :(

File: reports/qc/script-for-analysis-runner.sh (hunk 2 above)
Comment (vladsavelyev) [reply reports/qc/script-for-analysis-runner.sh:46]: Remember I tried to figure out why I can't make the plots bigger when rendered in the command line into an HTML? That's the reason. The only way I found to make them approximately the same size in Rstuido and in HTML is to do this sed :(

File: reports/qc/qc.Rmd (hunk 3 above)
Comment (vladsavelyev) [reply reports/qc/qc.Rmd]: Thank you! :)

File: reports/qc/script-for-analysis-runner.sh (hunk 2 above)
Comment (pdiakumis) [reply reports/qc/script-for-analysis-runner.sh:46]: Oh dear ;-) All good.
I don't rely on RStudio's viewer when comparing plot sizes; I always just look at the final HTML. Plus on a 13" Mac it's sometimes pointless looking at them in the viewer.

File: reports/qc/qc.Rmd [hunk 4]
```diff
@@ -326,9 +346,11 @@ p_data %>%
   xlab("")
 ```
 
+Table listing all samples {nrow(all_data)} samples in {n_batches} batches.
```
Comment (pdiakumis) [inline reports/qc/qc.Rmd]:
```suggestion
Table listing all {nrow(all_data)} samples in {n_batches} batches.
```

File: reports/qc/qc.Rmd [hunk 5]
```diff
@@ -489,28 +540,45 @@ p_data %>%
   theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
         panel.grid.major.x = element_blank(),
         panel.grid.minor.x = element_blank()) +
-  facet_wrap(~lane) +
-  ggtitle(glue("QC metrics (for {length(unique(p_data$sample))} samples).")) +
-  xlab("")
+  facet_wrap(~flowcell1) +
+  ggtitle(glue("Duplication rate groupped by flowcell, colored by batch")) +
```
Comment (pdiakumis) [inline reports/qc/qc.Rmd]:
```suggestion
  ggtitle(glue("Duplication rate grouped by flowcell, colored by batch")) +
```

**Review by pdiakumis — APPROVED** [review]

File: reports/qc/qc.Rmd (hunk 4 above)
Comment (vladsavelyev) [reply reports/qc/qc.Rmd]: Forgot to merge these suggestions! Sorry and thanks for fixing the typos.

**Pull request merged**